au4/src/effects/builtin/tonegen/ToneView.qml

```diff
@@ -1,7 +1,130 @@
-import QtQuick 2.15
+import QtQuick
+import QtQuick.Controls
+import Muse.UiComponents
+import Audacity.Effects
+import Audacity.Playback
+import "../common"
 
-Rectangle {
+EffectBase {
 
-    color: "#009090"
+    property string title: "Tone" // review
+    property alias instanceId: tone.instanceId
 
+    property bool isApplyAllowed: true
+
+    width: 300
+    height: 200
+
+    ToneViewModel {
+        id: tone
+        onSampleRateChanged: {
+            timecode.sampleRate = tone.sampleRate
+        }
+    }
+
+    Component.onCompleted: {
+        tone.init()
+    }
+
+    Column {
+        anchors.fill: parent
+        spacing: 16
+
+        Row {
+
+            anchors.horizontalCenter: parent.horizontalCenter
+
+            spacing: 4
+
+            StyledTextLabel {
+                anchors.verticalCenter: parent.verticalCenter
+                text: qsTrc("effects/tone", "Waveform:")
+            }
+
+            ComboBox {
+                anchors.verticalCenter: parent.verticalCenter
+                width: 80
+
+                model: ["Sine", "Square", "Sawtooth", "Square, no alias", "Triangle"]
```

Comment:
It is better to return this list (model) from the view model and somehow coordinate it with the enumerator.